crates/ghost_actor/src/ghost_system.rs

```diff
@@ -0,0 +1,284 @@
+use crate::*;
+use std::sync::Arc;
+
+/// struct used for hinting on whether / when to next run this process fn
+pub struct GhostProcessInstructions {
+    should_continue: bool,
+    next_run_delay_ms: u64,
+}
+
+impl Default for GhostProcessInstructions {
+    fn default() -> Self {
+        Self {
+            should_continue: false,
+            next_run_delay_ms: 0,
+        }
+    }
+}
+
+impl GhostProcessInstructions {
+    pub fn should_continue(&self) -> bool {
+        self.should_continue
+    }
+
+    pub fn set_should_continue(mut self, should_continue: bool) -> Self {
+        self.should_continue = should_continue;
+        self
+    }
+
+    pub fn next_run_delay_ms(&self) -> u64 {
+        self.next_run_delay_ms
+    }
+
+    pub fn set_next_run_delay_ms(mut self, next_run_delay_ms: u64) -> Self {
+        self.next_run_delay_ms = next_run_delay_ms;
+        self
+    }
```

Comment:
Do you really need mutating "instructions" ?

Reply:
This just allows a builder pattern, so when we're using it, we can do things like:

```rust
return Ok(GhostProcessInstructions::default().set_next_run_delay_ms(50))
```